src/Reducer/UserDataReducer.js

```diff
@@ -15,6 +15,21 @@ export function UserDataReducer(state, action) {
         JSON.stringify(new_liked_video_after_remove)
       );
       return { ...state, liked_video: new_liked_video_after_remove };
+    case "ADD_TO_WATCH_LATER":
+      const new_watch_later_after_add = action.payload;
+      localStorage.setItem(
+        "my_watch_later_video",
+        JSON.stringify(new_watch_later_after_add)
+      );
+      return { ...state, watch_later: new_watch_later_after_add };
+
+    case "REMOVE_FROM_WATCH_LATER":
+      const new_watch_later_after_remove = action.payload;
+      localStorage.setItem(
+        "my_watch_later_video",
+        JSON.stringify(new_watch_later_after_remove)
+      );
+      return { ...state, watch_later: new_watch_later_after_remove };
     case "RESET":
       return { liked_video: [], watch_later: [], playlist: [] };
```

Comment:
you can just return the initial state to reset.

Reply:
For the reset i am erasing all the data from context..
Using this event when user logs out.